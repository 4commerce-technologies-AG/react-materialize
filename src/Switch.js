import React from 'react';
import PropTypes from 'prop-types';
import idgen from './idgen';

const Switch = ({
  id,
  checked,
  onChange,
  disabled,
  offLabel,
  onLabel,
  ...props
}) => {
  return (
    <div className="switch">
      <label htmlFor={id}>
        {offLabel}
        <input
          id={id}
          disabled={disabled}
          onChange={onChange}
          type="checkbox"
          checked={checked}
          {...props}
        />
        <span className="lever" />
        {onLabel}
      </label>
    </div>
  );
};

Switch.propTypes = {
  /*
   * override id
   * @default idgen()
   */
  id: PropTypes.string.isRequired,
  className: PropTypes.string,
  /*
   * label for off
   */
  offLabel: PropTypes.string.isRequired,
  /*
   * label for on
   */
  onLabel: PropTypes.string.isRequired,
  /*
   * onChange callback
   */
  onChange: PropTypes.func,
  /*
   * disabled input
   */
  disabled: PropTypes.bool,
  /*
   * initialise checkbox checked
   */
  checked: PropTypes.bool
};

Switch.defaultProps = {
  id: `switch-${idgen()}`
};

export default Switch;
