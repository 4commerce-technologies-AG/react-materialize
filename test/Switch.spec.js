import React from 'react';
import { shallow } from 'enzyme';
import Switch from '../src/Switch';

describe('<Switch />', () => {
  test('renders', () => {
    const wrapper = <Switch offLabel="Off" onLabel="On" />;
    expect(shallow(wrapper)).toMatchSnapshot();
  });
});
