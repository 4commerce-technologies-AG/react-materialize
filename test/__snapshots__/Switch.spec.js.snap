// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch /> renders 1`] = `
<div
  className="switch"
>
  <label
    htmlFor="switch-mockId"
  >
    Off
    <input
      id="switch-mockId"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="lever"
    />
    On
  </label>
</div>
`;
